extensions/amp-video/0.1/video-cache.js

```diff
@@ -173,12 +176,34 @@ function applyCaptionsTrackToVideo(videoEl, captionsResponse) {
   ) {
     return;
   }
-  const trackEl = createElementWithAttributes(videoEl.ownerDocument, 'track', {
-    'src': captionsResponse['src'],
-    'srclang': captionsResponse['srclang'],
-    'kind': 'captions',
-  });
+
+  Services.extensionsFor(ampdoc.win).installExtensionForDoc(
```

Comment:
XHR calls to the Google video cache are made from videos `buildCallback`, so they happen on page load.
If you write this code here, the extension will load on page load, even if the captions are on page 10.

Instead, maybe you could lazy-load `amp-story-captions.js` from `amp-video.setUpCaptions_()`, which runs in the `layoutCallback` and will work great with our existing lazy loading/distance system.